libyul/backends/wasm/WasmDialect.h

```diff
@@ -56,7 +56,14 @@ struct WasmDialect: public Dialect
 private:
 	void addEthereumExternals();
 
-	void addFunction(std::string _name, size_t _params, size_t _returns, bool _movable = true, bool _literalArguments = false);
+	void addFunction(
```

Comment:
Might it make sense to have this take two vectors of types instead for arguments and return values?

Or maybe the whole function could just be ``add(BuiltinFunction x) { YulString name = x.name; m_functions[name] = std::move(x); }`` instead and be called as ``add(BuiltinFunction{ ... })``? It takes almost all elements in ``BuiltinFunction`` as arguments anyways right now - and those it doesn't are set manually above most of the time...

Reply:
Yep, I will change it to take a vector of params and retparams